open-craft/frontend-lib-content-components
# [FAL-3383] Implement new video UX flow on new video editor

Temporal PR to see the changes until https://github.com/openedx/frontend-lib-content-components/pull/278 will be merged. You can see the changes here for a more acurated review.

Internal Jira task: [FAL-3383](https://tasks.opencraft.com/browse/FAL-3383)

**Testing instructions**

- Use this branch of [edx-platform](https://github.com/openedx/edx-platform/pull/32024) and this branch of [frontend-app-course-authoring](https://github.com/open-craft/frontend-app-course-authoring/pull/1) 
- Login on studio with a super user.
- We need to test the flow before to create any video.
- On a course create a unit and create a new video. Verify that it redirects to the video gallery, but after loading, it redirects to the video upload page. 
- Go to [Django Admin](http://localhost:18010/admin/edxval/coursevideo/) and create a conection with a video and the Demo course.
    - Edit the duration of the videos.
- On a course create a unit and create a new video. Verify that it redirects to the video gallery.
- Check the cancel button
- Create new video and on the video gallery, select a video and go to the next page.
- Verify that the Video ID and de duration are filled.
- Click on the `Replace video button`.

To test the video upload, you need to configure an S3, but there is an easy way to test this
- Read [this comment](https://github.com/open-craft/frontend-lib-content-components/pull/8/files#diff-8c87d7bec267a780f66f8210f786b97cec0d8bdafacc162a220d13343393c286R30)
- Update the [videos_handler](https://github.com/openedx/edx-platform/blob/d622db59b7bfe2e4c11fc11b7cb2d475cfda551b/cms/djangoapps/contentstore/views/videos.py#L188) function on edx-paltform to return overwritten data. You can use this [gist as reference](https://gist.github.com/ChrisChV/e662a5bdb4145a80606295f5b677d5b5)
- Create a video and on the video gallery, click on the upload button.
- Upload any video
- Verify that the Video ID and de duration are filled.

**Author notes**

- The video upload only works with an S3 configuration.
- I added a temporal line to test the video upload without an S3 configuration: [Comment](https://github.com/open-craft/frontend-lib-content-components/pull/8/files#diff-8c87d7bec267a780f66f8210f786b97cec0d8bdafacc162a220d13343393c286R30)
- It's missing to auto fill the transcripts from the selected video

**Release notes**

- **IMPORTANT:** Delete [this line](https://github.com/open-craft/frontend-lib-content-components/pull/8/files#diff-8c87d7bec267a780f66f8210f786b97cec0d8bdafacc162a220d13343393c286R30) before merge
